src/backend/mi2/mi2.ts

```diff
@@ -852,15 +852,13 @@ export class MI2 extends EventEmitter implements IBackend {
 		if (trace)
 			this.log("stderr", "getRegisters");
 
-		// Getting register names and values are separate GDB commands.
-		// We first retrieve the register names and then the values.
-		// The register names should never change, so we could cache and reuse them,
-		// but for now we just retrieve them every time to keep it simple.
-		const names = await this.getRegisterNames();
+		if (!this.regNames) {
+			this.regNames = await this.getRegisterNames();
```

Comment:
the idea is good, but we need to store the regnames per inferior (you may switch between multiple ones via UI), not globally, as the inferiors can have different registers (example: server in 64 bit and client in 32 bit (or vice versa), attached to both)